File: vxpt-bbs/src/main/kotlin/com/github/blanexie/vxpt/bbs/user/meta/entity/UserDO.java
package com.github.blanexie.vxpt.bbs.user.meta.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

/**
 * 2023/2/10 3:47 下午
 *
 * @author xiezhicheng
 */
@Data
@TableName("user")
public class UserDO {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String email;
    private String nickName;
    private String password;
    private String coverImg;
    private Integer sex;
    private String role;
    private Integer status;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}

File: vxpt-bbs/src/main/java/com/github/blanexie/vxpt/bbs/user/meta/entity/UserDO.kt
package com.github.blanexie.vxpt.bbs.user.meta.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.annotation.TableName
import java.time.LocalDateTime

@TableName("user")
class UserDO(
    @TableId(type = IdType.AUTO)
    var id: Long?,
    var email: String,
    var nickName: String,
    var password: String,
    var coverImg: String?,
    var role: String,
    var sex: Int,
    var status: Int,
    var createTime: LocalDateTime,
    var updateTime: LocalDateTime,
) {
}